include/rcpputils/compile_warnings.hpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2025, cellumation GmbH
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//    * Redistributions of source code must retain the above copyright
+//      notice, this list of conditions and the following disclaimer.
+//
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+//
+//    * Neither the name of the copyright holder nor the names of its
+//      contributors may be used to endorse or promote products derived from
+//      this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+#ifndef RCPPUTILS__COMPILE_WARNINGS_HPP_
+#define RCPPUTILS__COMPILE_WARNINGS_HPP_
+
+#if !defined(_WIN32)
+#define RCPPUTILS_DEPRECATION_WARNING_OFF_START \
+  _Pragma("GCC diagnostic push") \
+  _Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")
+#else  // !defined(_WIN32)
+#define ROS_DEPRECATION_WARNING_OFF_START \
+  _Pragma("warning(push)") \
+  _Pragma("warning(disable: 4996)")
+#endif
+
+#if !defined(_WIN32)
+#define RCPPUTILS_DEPRECATION_WARNING_OFF_STOP \
+  _Pragma("GCC diagnostic pop")
+#else  // !defined(_WIN32)
+#define ROS_DEPRECATION_WARNING_OFF_STOP \
+  _Pragma("warning(pop)")
+
+#endif
```

Comment:
one less `if` statement for the macro definition like below?

```suggestion
#if !defined(_WIN32)
#define RCPPUTILS_DEPRECATION_WARNING_OFF_START \
  _Pragma("GCC diagnostic push") \
  _Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")
#define RCPPUTILS_DEPRECATION_WARNING_OFF_STOP \
  _Pragma("GCC diagnostic pop")
#else  // !defined(_WIN32)
#define ROS_DEPRECATION_WARNING_OFF_START \
  _Pragma("warning(push)") \
  _Pragma("warning(disable: 4996)")
#define ROS_DEPRECATION_WARNING_OFF_STOP \
  _Pragma("warning(pop)")
#endif
```